# **Fundamentos de Estadística**

---
---
<!-- Star Wars: Episodio II - El ataque de los clones -->


<!-- script html for image -->
<figure>
<center>

<img src='https://upload.wikimedia.org/wikipedia/commons/5/52/Sw-ep2-logo.png' width="500" height="150" />

</figure> </center>


Fuente de la imagen: [https://es.wikipedia.org](https://es.wikipedia.org/wiki/Star_Wars:_Episodio_II_-_El_ataque_de_los_clones)


# **Índice**

---






> [Fundamentos de Estadística](#scrollTo=qRNdbPLqB7hQ&line=6&uniqifier=1)
<br>
>>
>> [1 - Introducción a la Estadística](#scrollTo=rSvsnuw_n2p6&line=21&uniqifier=1) 
>>
>>>[1.1. Concepto de estadística](#scrollTo=rSvsnuw_n2p6&line=1&uniqifier=1) \\
>>>
>>>[1.2. Conceptos básicos](#scrollTo=GQwDTSuFR3te&line=1&uniqifier=1) \\
>>>> [1.2.1. Población](#scrollTo=LKI4dVI636i8&line=1&uniqifier=1) \\
>>>> [1.2.2. Muestra](#scrollTo=rJgxxjXRxvgy&line=2&uniqifier=1) \\
>>>> [1.2.3. El Dato](#scrollTo=sisv_pAtHMmm) \\
>>>> [1.2.4. Variable estadística](#scrollTo=2XnUyIIzDZOc&line=1&uniqifier=1) \\
>>>
>>>[1.3. Manipulación básica de los datos: explorar el _dataset_](#scrollTo=HahUx1Iddj1X) \\
>>>
>>>
>>[Actividad: Titanic](#scrollTo=jF6YiKGfYIj0&line=1&uniqifier=1) \\
>>> [Solución](#scrollTo=ePluH30VL_Ku) 
>>
>> [Ideas Clave](#scrollTo=1g2mu1uOIB5Q&line=4&uniqifier=1) 

<br>

< [Índice Global](https://colab.research.google.com/drive/1yCeM4hAVmT-GSm8L-Zgc0u02t1pUTSxn?usp=sharing) | 
[2 - Estadística Descriptiva Unidimensional](https://colab.research.google.com/drive/1jSz5DRRuCg7quG0rLWXiJz79vs-sMT_o?usp=sharing) >

# 1 - Introducción a la Estadística

## 1.1. Concepto de estadística

La primera cuestión para entender *estadística* es preguntarnos: **¿Qué es la estadística?**

Si miramos en la RAE, [estadística](https://dle.rae.es/estad%C3%ADstico) es el << estudio de los datos cuantitativos de la población, de los recursos naturales e industriales, del tráfico o de cualquier otra manifestación de las sociedades humanas >>.

También encontramos la definición de << rama de la matemática que utiliza grandes conjuntos de datos numéricos para obtener inferencias basadas en el cálculo de probabilidades >>.

De una forma un poco más tosca, se pude decir que los datos cuantitativos **son aquellos datos que se pueden contar**.

Un dicho que se puede usar para describir la necesidad de aplicar estadística en un estudio científico es que **"lo que no son cuentas, son cuentos"**. Esto se podría interpretar como que lo que no se puede cuantificar o medir se convierte en una idea o intuición, por lo que no es lo mismo que tener una justificación o estudio científico de un fenómeno o suceso. 
  
Si nos vamos a *Wikipedia* y buscamos [historia de estadística](https://es.wikipedia.org/wiki/Historia_de_la_estad%C3%ADstica), se podría definir como la << disciplina ocupada de recolectar, resumir y analizar los datos >>. Por tanto, la estadística se basa simplemente en tomar nota para cuantificar y **buscar para entender** cualquier objeto de estudio o fenómeno. 

A veces remontarse al pasado nos hace entender mejor los conceptos presentes. Esta vez recurrimos a una *fuente oficial* para explicar los origenes. El **Instituto Nacional de Estadística** (INE) tiene un canal donde abarca la historia de la estadística desde la *Edad Antigua* hasta la *Edad Contemporánea*. El enlace es el siguiente:

[www.ine.es/explica](https://www.ine.es/explica/explica_historia_estadistica.htm)
  
En definitiva, el nombre de Estadística se refiere al interés de esta rama de las Matemáticas en asuntos del Estado: censos de poblaciones, empadronamiento, índice de natalidad, de mortalidad, etc. 

Actualmente, la Estadística interviene en los campos más diversos y su introducción tanto en el mundo científico como empresarial, se debe a la importancia indiscutible para el desarrollo de todas las ciencias (Psicología, Medicina, Economía, Historia, Física, Química, etc.).

Como curiosidad, comentar que uno de los primeros métodos que se usó para recolectar, resumir y analizar los datos en la historia de las matemáticas fue el **ábaco**, el cual permite realizar operaciones aritméticas sencillas​ y otras más complejas y cuyo origen real sigue siendo un misterio.

<!-- script html for image -->
<figure>
<center>
<img src='https://p1.pxfuel.com/preview/201/199/754/abacus-computational-aids-wooden-balls-count.jpg' width="400" height="300" />
<figcaption>Imagen1. Ábaco</figcaption></center>
</figure>

Fuente de la imagen: [www.p1.pxfuel.com](https://pixabay.com/es/photos/%C3%A1baco-ayudas-de-c%C3%B3mputo-7935/)





<!-- script html for "Recuerda" -->
<!-- espaciado salto <br> espaciado de linea &nbsp; -->
<br>
<p> <mark> PIENSA UN MINUTO </mark> </p>
<hr>
<p>
<br>
<center><b>

¿ Porqué crees que es tan importante la estadística ?

</center></b>
<br>

<!-- script html for "Recuerda" -->
<!-- espaciado salto <br> espaciado de linea &nbsp; -->
<br>
<p> <mark> RECUERDA</mark> </p>
<hr>
<p>

La estadística es el área de las matemáticas que <strong>recuenta, agrupa y extrae información </strong> de los datos.

Esa información extraída ayuda a entender la realidad y a poder tomar decisiones en consecuencia.</p>

<p>Dicho en clave de humor: <strong>si sabes contar,</strong> contar literalmente 1, 2, 3, etc. y <strong>si sabes sumar,</strong> agrupar o resumir datos entonces, <strong>sabes estadística.</strong> Así de sencilla y fácil es la estadística. </p>
<br>

> **Todo aquello que es susceptible de ser medido** (contado) o que se mide, entonces, **se puede analizar con estadística**. 
>
> Quizás por eso es preocupante que haya datos, que aun siendo conocidos (a través de medios de comunicación como periódicos o televisión) <u>no se tomen los datos para analizar el grado de impacto real en la sociedad.</u>

## 1.2. Conceptos básicos

  Para entender los conceptos de los siguientes temas, primero hay que comprender toda la terminología usada en estadística.

###  1.2.1.  Población

  Es el **conjunto de todos los elementos cuyas propiedades se van a estudiar**. En algunos sitios recibe el nombre de *Universo*. Existen dos categorías:

  1.   **Población finita**. Cuando se puede determinar a priori la cantidad exacta de elementos u observaciones. Por ejemplo: 

    *   Número exacto de universidades que hay en España.
    *   Número exacto de trabajadores en una empresa o compañía.
    *   Número de hijos que tiene una familia.

  2.   **Población infinita**. Cuando no se puede determinar a priori una cantidad exacta o es imposible de determinar. Por ejemplo:

    *   Número de estrellas en el universo.
    *   Número de granos de arena en el mar.
    *   Número de insectos a nivel mundial.

<!-- script html for image -->
<figure>
<center>
<img src='https://es-static.z-dn.net/files/d9f/0bd98686b067cfcc3bb585dc907a6415.png' width="600" height="300" />
<figcaption>Imagen2. Población</figcaption></center>
</figure>

Fuente de la imagen: [www.portafoliosegundoparcialestadistica.blogspot.com](http://portafoliosegundoparcialestadistica.blogspot.com/2017/04/poblacion-finita-e-infinita.html)

### 1.2.2. Muestra

En muchas ocasiones, no se puede trabajar con toda la población, por lo que se usa un **subconjunto de la población**. Esto es, **una muestra representativa de la misma** y, para lograrlo hay que usar criterios y técnicas de muestreo. 

Como norma general, **una** **muestra representativa** **debe reflejar todas las características de la población**.

Por ejemplo, si se desea conocer el resultado de las próximas elecciones de un país, resulta poco viable preguntar a toda la población de ese país. Por lo que se opta por obtener un subconjunto de esta población, asumiendo un margen de error, a partir del cual se podrán sacar conclusiones sobre la votación de toda la población.

Existen **dos clasificaciones** para la muestra estadística: 
<br>
  1. **Probabilística**.Todos los elementos de la población tienen la misma posibilidad de formar parte de ella. El azar interviene de alguna manera.
Por Ejemplo:

    *   Todas las personas enfermas de un hospital, sin importar la enfermedad o las patologías que presente.

    *   Todo el alumnado de una Universidad, sin importar la nota.


  2. **No probabilística**. No todos los elementos de la población tienen la misma posibilidad de formar parte de la muestra. No se usa el azar, sino el criterio del investigador, es decir, se decide si la muestra es o no representativa. Por ejemplo:
    *   Todas las personas enfermas de un hospital que presentan una misma enfermedad (Covid) o patología (diabetes). Las personas que no presentan esa patología no son incorporadas en el estudio.

    *   Todo el alumnado de una Universidad que suspende o abandona. El estudio quiere determinar las causas de suspenso o abandono y no se incorporan los alumnos/as con aprobados.

\\
\begin{equation} 
\text {Tipos de Muestreo} = 
     \begin{cases}
      \text{Probabilístico} &\quad\text{Misma probabilidad }\\ \\
      \text{No Probabilístico} &\quad\text{Distinta probabilidad}\\
     \end{cases}\\     
\end{equation}

\\
<p> <mark>NOTA</mark> </p>
<hr>
Hay distintas tipologías dentro del muestreo que se desarrollarán más adelante.

En consecuencia, **muestra y población son conceptos relativos**.

Una **población** es ***un todo*** y una **muestra** es ***una fracción o segmento*** de ese todo.

<!-- script html for image -->
<figure>
<center>
<img src='https://economipedia.com/wp-content/uploads/Poblaci%C3%B3n-estad%C3%ADstica.png' width="600" height="300" />
<figcaption>Imagen3. Población y Muestra </figcaption></center>
</figure>

Fuente de la imagen: [www.economipedia.com](https://economipedia.com/definiciones/poblacion-estadistica.html)


### 1.2.3. El Dato

**DEFINICIÓN DE DATO**

Cada uno de los individuos, cosas, entes abstractos que **integran una población o universo determinado**.

Dicho de otra forma, cada valor observado de la variable.

Por ejemplo:

Se pueden estudiar personas, animales, enfermedades, temperaturas, bombillas, ordenadores, etc.


Cualquier estudio estadístico comienza con **la recogida de datos**. Esta recogida **puede ser física y directa** (aparatos de medición, encuestas, etc.) **o virtual** mediante la consulta de información a través de un programa.

Los datos pueden proceder de distintas instituciones u organismos públicos, de la navegación por Internet, de la propia compañía en que trabajemos o de datos recogidos a través de encuestas.

El objetivo de la recogida de datos, es poder analizarlos posteriormente. Por tanto, hay que procurar, en la medida de lo posible, hacer que **la presentación de estos datos sea de forma sencilla, coherente y atractiva** para el consumidor final. 

En este sentido, la estadística dispone los datos, generalmente, en tablas y, en muchas ocasiones, se ayuda a su vez de gráficos que pretenden aclarar aspectos reseñables de los datos recogidos.

### 1.2.4  Variable estadística

La variable estadística es una <strong> característica o cualidad de un individuo que se pueden medir, y puede adquirir diferentes valores. </strong>

Por ejemplo, el color de los ojos, la estatura de una persona, el grupo sanguíneo o las notas obtenidas de una prueba.

Existen dos clasificaciones para las variables estadísticas: 
<br>
  1. **Cuantitativas o Numéricas**. Como su propio nombre indica, son variables que se pueden [cuantificar](https://definicion.de/cuantitativo/), es decir, se pueden medir. Son de tipo numérico y, por tanto, se representan numéricamente. Dentro de esta clasificación hay dos tipologías:

    1.1 **Variables Discretas:** utilizan valores enteros y finitos. Son valores numéricos indivisibles.
    Por ejemplo: número de hijos, número de coches, número de viviendas, etc. 
    
    Su representación matemática es el conjunto de los números naturales: 
   
    $\quad \mathbb N = \{  1, 2, 3, 4,.. \} \\ $  

    1.2 **Variables Continuas:** utilizan valores objetivos y pueden tomar un número de valores infinitos. Se caracterizan por utilizar valores decimales (divisibles). Ejemplos: la altura, el peso, el dinero, etc.

  Su representación matemática es el conjunto de los números racionales: 
  
  $\quad \mathbb Q =  \{ a \div b \quad donde  \quad  a, b \in \mathbb Z \quad y \quad b \neq 0 \} = \{  1.3, 2 , 2.578, 3.275, 4,... \}  $

\\
<h4><center> Variables Cuantitativas </center></h4>
<table class="var" cellspacing="0">
<tr class="var">
<th width="20%" class="var" scope="col">Tipos</th>
<th width="45%" class="var" scope="col">Definición</th>
<th width="35%" class="var" scope="col">Ejemplos</th>
</tr>
<tr class="var">
<th width="20%" class="var1">Discreta</th>
<td width="45%" class="var2"> La variable solo puede tomar valores en número determinado de valores. <br /> En cada intervalo de valores la variable solo puede tomar un valor.</td>
<td width="35%" class="var3">&#8211; Canastas en un partido (20; 21; 22; pero no 21,5). <br /> &#8211; Hijos por familia (0, 1, 2, 3,&#8230;). </td>
</tr>
<tr class="var">
<th width="20%" class="var1">Continua</th>
<td width="45%" class="var2"> La variable puede adquirir cualquier valor dentro de un intervalo de valores determinado. </td>
<td width="35%" class="var3">&#8211; Peso (53,53&nbsp;kg; 89,4&nbsp;kg,&#8230;).</td>
</tr>
</table>

\\

  2. **Cualitativas o Categóricas**. Como su propio nombre indica, son variables que están vinculadas a una cualidad. Es decir, las variables [cualitativas](https://definicion.de/cualitativo/) no pueden ser calculadas con números, sino que son clasificadas con palabras. Dentro de esta clasificación hay tres tipologías:

    2.1 **Cualitativas ordinales:** son aquellas que siguen un orden o una jerarquía. Por ejemplo, el nivel socioeconómico (alto, medio o bajo), la nota de un examen (suspenso, aprobado, bien, notable, sobresaliente), potencia de un aparato o pulsaciones de un corazón (leve, moderado, fuerte).

    2.2 **Cualitativas nominales:** se trata de aquellas variables que no siguen ningún orden en específico. Por ejemplo, el grupo sanguíneo, el color de pelo, el color de los ojos, etc.

    2.3 **Cualitativa binaria o dicotómica:** es un caso particular de variable nominal con solo dos categorías o dos resultados posibles. 
    
    Si las dos categorías determinan dos estados cualesquiera se denomina <em><strong>binaria simétrica</strong></em> (por ejemplo: sexo). Si el 1 determina la presencia de una característica y el 0 su ausencia (como por ejemplo: enfermedad, trabajo, etc.), la variable se denomina <em><strong>binaria asimétrica</strong></em>.</td>


\\
<h4><center> Variables Cualitativas </center></h4>
<table class="var" cellspacing="0">
<tr class="var">
<th width="20%" class="var" scope="col">Tipos</th>
<th width="45%" class="var" scope="col">Definición</th>
<th width="35%" class="var" scope="col">Ejemplos</th>
</tr>
<tr class="var">
<th width="20%" class="var1">Nominal</th>
<td width="45%" class="var2"> Son variables cualitativas cuyas categorías no siguen ningún orden.</td>
<td width="35%" class="var3">&#8211; Color (blanco, rojo, azul,&#8230;).<br />&#8211; Lateralidad (zurdo, diestro). </td>
</tr>
<tr class="var">
<th width="20%" class="var1">Ordinal</th>
<td width="45%" class="var2"> Son las variables categóricas con orden o jerarquía. </td>
<td width="35%" class="var3">&#8211; Nota examen. <br /> (suspenso, aprobado, notable, sobresaliente). <br /> &#8211; Nivel económico. <br /> (pobre, clase media, rico). <br /> &#8211; Medalla deportiva. <br /> (Oro, plata, bronce).</td>
</tr>
<tr class="var">
<th width="20%" class="var1">Binaria o dicotómica</th>
<td width="45%" class="var2"> Es un caso particular de la variable nominal. Son sólo dos categorías.</td>
<td width="35%" class="var3">&#8211; Sexo (mujer, hombre). Simétrica. <br />&#8211; Enfermo (si, no). Asimétrica. </td>
</tr>
</table>

\\

**ESQUEMA RESUMEN TIPOS DE VARIABLES**
<br>

\\
\begin{equation} 
Variables = 
     \begin{cases}
      \text{Cuantitativas o Numéricas} \begin{cases}
                             \text{Discreta}\\ \\
                             \text{Continua} \\
                           \end{cases}\\   \\
      \text{Cualitativas o Categóricas} \begin{cases}
                             \text{Nominal}\\ \\
                             \text{Ordinal} \\ \\
                             \text{Binaria o Dicotómica}\\
                           \end{cases}   \\ 
     \end{cases}\\     
\end{equation}


<h3><strong>Tipos de variables en el análisis de datos</h3></strong>

En función del análisis que se realice, **las variables pueden recogerse de manera numérica pero ser categóricas**.

Por ejemplo, la variable fecha es una variable con la que se toma el dato (y se almacena) de manera numérica pero no es una variable cuantitativa ya que no tiene sentido sumar literalmente los años (por ejemplo: 2020+2021=4041). 

La variable como tal, es una variable recogida o expresada de forma numérica (2020) cuyo uso es categórico, y sirve, por ejemplo, para clasificar la información sobre el año en que ocurrió un suceso.

Otro ejemplo podrían ser los códigos postales, que se almacena en forma de código numérico para hacer una consulta ágil de la información, aunque desde el punto de vista analítico sirve para clasificar ya que es el equivalente a los nombres de las poblaciones a las que haga referencia. Por ejemplo: el 28020 (corresponde a Madrid) y el 08300 (equivaldría a poner Barcelona).

En función del análisis de datos, existen **dos tipos de variables** que son:

*   **Variables categóricas o de clasificación**: también se denominan variables cualitativas o variables de atributos. Los valores de una variable categórica son **categorías o grupos mutuamente excluyentes**. Los datos categóricos pueden tener o no tener un orden lógico. Por ejemplo:

  *   Zonas geográficas (regiones, provincias, etc.) de una ciudad.
  *   Clasificación del tipo de cliente.
  *   Tipos de vehículo (gasolina, diésel, eléctrico, etc.).
   

*   **Variables analíticas o cuantitativas:** son números que suelen representar una medición (frecuencia o conteo, sumas, etc.). Por ejemplo:

  *   Importe de unas facturas.
  *   Número de veces que ocurre una enfermedad.
  *   Coste de mantenimiento de un hogar.

Este tipo de variables, a veces, necesita un tratamiento y no se puede trabajar con el dato "en bruto". Un ejemplo suele ser "la edad" cuyo valor en sí no aporta mucha información de forma individual. Sin embargo, aporta mucha información si se hace por rangos o mediante una clasificación. Por ejemplo: niños [0-18 años), adultos [18-65 años) y mayores [>65 años).


**RESUMEN TIPOS DE VARIABLES**

\\
\begin{equation} 
Variables = 
     \begin{cases}
      \text{Categórica o de Clasificación}   \\
      \\ 
      \text{Analítica o Cuantitativa}   \\ 
     \end{cases}\\     
\end{equation}

\\

<p> <mark>IMPORTANTE</mark> </p>
<hr>

Las variables **binarias o dicotómicas** pueden ser tratadas, a su vez, de forma **categórica y analítica** (teniendo en cuenta que sólo se puede medir la frecuencia en este apartado).


<!-- script html for "Recuerda" -->
<!-- espaciado salto <br> espaciado de linea &nbsp; -->
<p> <mark>RECUERDA</mark> </p>
<hr>

*   **La variable estadística** es la que después **se relacionará con algún problema o fenómeno, que se desea investigar**, entender su causa o fenómenos asociados y/o buscar posibles soluciones.
*   El **primer paso de cualquier tipo de análisis** radica en **estudiar el tipo de variables** de cualquier dataset.
*   Una **variable estadística es una cualidad o cantidad** que poseen los individuos u objetos de estudio de una población de análisis.

## 1.3. Manipulación básica de los datos: explorar el _dataset_

Para poder describir el dataset, se introducirá el uso de la librería **pandas**, que es un paquete de **Python** que provee de una serie de funciones, métodos y estructuras de datos para trabajar los dataset de forma fácil e intuitiva.

**Pandas está construido sobre la librería de numpy**. Esto produce que algunos métodos o funciones de pandas no retornen tipos de datos creados en la propia librería, como los _DataFrame_ o _Series_, sino que devuelva estructuras de numpy como el _numpy.ndarray_. Al mismo tiempo, pandas puede cargar, fácilmente, matrices que provengan de numpy.

Para empezar a trabajar, lo primero que necesitamos es poder cargar el dataset con pandas y, luego, lograr obtener alguna información sobre sus columnas. Con esta finalidad, se abrirá un fichero en formato .csv para, posteriormente, analizar de qué tipo son las variables.

La sentencia de comandos para importar las librerías con las que trabajaremos es ***import*** seguida del paquete, al que se le asocia un alias para mayor comodidad. Esto es:

In [ ]:
# Importing the libraries
import pandas as pd
import numpy as np

## Actividad: Titanic

El **objetivo** de esta actividad consiste en **identificar el tipo y el subtipo de variables** que corresponda.

Antes de empezar a procesar, analizar y visualizar los datos, es necesario **entender el dataset**. La fuente del _dataset_ es de Kaggle, donde se facilita un archivo en formato CSV, con información del famoso accidente del Titanic.

También, se proporciona la información del diccionario de datos que se muestra a continuación:

<h3><center>Data Dictionary</center></h3>
<table>
<tbody>
<tr><th><b>Variable</b></th><th><b>Definition</b></th><th><b>Key</b></th></tr>
<tr>
<td>survival</td>
<td>Survival</td>
<td>0 = No, 1 = Yes</td>
</tr>
<tr>
<td>pclass</td>
<td>Ticket class</td>
<td>Nivel Socieconómico: 1 = 1st (upper), 2 = 2nd (middle), 3 = 3rd (lower)</td>
</tr>
<tr>
<td>sex</td>
<td>Sex</td>
<td></td>
</tr>
<tr>
<td>Age</td>
<td>Age in years</td>
<td> Age is fractional if less than 1. If the age is estimated, is it in the form of xx.5</td>
</tr>
<tr>
<td>sibsp</td>
<td># of siblings / número de hermanos, hermanas o marido/mujer</td>
<td></td>
</tr>
<tr>
<td>parch</td>
<td># of parents / número de parientes (padre, madre, hijos). 
</td>
<td>A algunos de los niños que viajaban sólo con la nanny se les puso la notación 0 para ellos.</td>
</tr>
<tr>
<td>ticket</td>
<td>Ticket number</td>
<td></td>
</tr>
<tr>
<td>fare</td>
<td>Passenger fare</td>
<td></td>
</tr>
<tr>
<td>cabin</td>
<td>Cabin number</td>
<td></td>
</tr>
<tr>
<td>embarked</td>
<td>Port of Embarkation</td>
<td>C = Cherbourg, Q = Queenstown, S = Southampton</td>
</tr>
</tbody>
</table>


### Solución 

[*A continuación, haz clic para conocer la solución.*]

**Abrir un archivo CSV** 

Para realizar tal acción se utiliza la función **read_csv** que retorna un DataFrame con la información del archivo. A continuación, vamos a cargar el archivo y aguardarlo en un _dataframe_. 

\\

<p> <mark>IMPORTANTE</mark> </p>
<hr>

Las columnas están separadas por comas (",").

In [ ]:
# Carga del fichero desde el enlace web y creación del dataframe
url = 'https://raw.githubusercontent.com/md-lorente/data/master/titanic.csv'

# Creacion Dataframe
df = pd.read_csv(url, sep=',')

# Visualización del dataframe (la cabecera)
df.head()

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0            1         0       3  ...   7.2500   NaN         S
1            2         1       1  ...  71.2833   C85         C
2            3         1       3  ...   7.9250   NaN         S
3            4         1       1  ...  53.1000  C123         S
4            5         0       3  ...   8.0500   NaN         S

[5 rows x 12 columns]

In [ ]:
# Resumen información del fichero
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB
None


Lo primero que debemos hacer al trabajar con un _dataset_ es analizar el tipo de variables con el que se trabajará:

1. **PassengerId:** es la identificación de cada uno de los pasajeros del Titanic. El código es único, consecutivo y entero. Es una variable **numérica de tipo discreto.**

2. **Survived:** es la identificación de si sobrevivió o no al accidente. Es una variable **cualitativa binaria o dicotómica asimétrica**.

3. **Pclass:** es una variable que mide el nivel socioeconómico. Se clasifica en 3 niveles: bajo, medio y alto. Por tanto, es una variable **cualitativa ordinal.**

4. **Name:** es una variable que recoge el nombre del pasajero. Es una variable **cualitativa nominal**.

5. **Sex:** es una variable que identifica el género del pasajero: hombre o mujer. Por tanto, es una variable **cualitativa binaria o dicotómica simétrica.**

6. **Age:** es la edad del pasajero. Cuando tiene decimales es una edad estimada (el decimal 0.5 indica que es una edad estimada). La variable edad, en general, es una variable cuantitativa discreta, pero tal y como se han tomado los datos es una variable **cuantitativa continua**. Nótese: que tiene decimales. 

7. **SibSp:** es el número de familiares: hermanos, hermanas o marido/esposa. Por tanto, se trata de una variable **cuantitativa discreta.**

8. **Parch:** es el número de familiares ascendientes o descendientes. Al igual que antes, se trata de una variable **cuantitativa discreta**.

9. **Ticket:** es el número del ticket del barco. Dado que el billete contiene números y letras, para identificar si tiene un orden establecido o no, nos fijamos en los datos que tienen una clasificación (empiezan por CA o PC seguido de un número secuencial). Por tanto, se trata de una variable **cualitativa ordinal.**

10. **Fare:** es la tarifa que paga cada pasajero. Es, por tanto, una variable **cuantitativa continua.**

11. **Cabin:** es el número de cabina, el cual tiene un orden de letra y número para clasificar las habitaciones dentro del barco. Es una variable **cualitativa ordinal**.

12. **Embarked:** es el puerto de embarque, en el que hay tres lugares donde se embarcaron los pasajeros: Cherbourg,  Queenstown y Southampton. No tiene un orden, por lo que se trata de una variable **cualitativa nominal.**



<h3> Un paso más allá... </h3>

De las variables analizadas, realizaremos una parada para analizar 3 variables:

*  Ticket.
*  Cabin.
*  Age.

<br>

<h4> <strong>Análisis de la variable Ticket </strong> </h4>


In [ ]:
# Análisis de la variable Ticket
billetes = pd.unique(df['Ticket'])
billetes.sort()
print(billetes)

['110152' '110413' '110465' '110564' '110813' '111240' '111320' '111361'
 '111369' '111426' '111427' '111428' '112050' '112052' '112053' '112058'
 '112059' '112277' '112379' '113028' '113043' '113050' '113051' '113055'
 '113056' '113059' '113501' '113503' '113505' '113509' '113510' '113514'
 '113572' '113760' '113767' '113773' '113776' '113781' '113783' '113784'
 '113786' '113787' '113788' '113789' '113792' '113794' '113796' '113798'
 '113800' '113803' '113804' '113806' '113807' '11668' '11751' '11752'
 '11753' '11755' '11765' '11767' '11769' '11771' '11774' '11813' '11967'
 '12233' '12460' '12749' '13049' '13213' '13214' '13502' '13507' '13509'
 '13567' '13568' '14311' '14312' '14313' '14973' '1601' '16966' '16988'
 '17421' '17453' '17463' '17464' '17465' '17466' '17474' '17764' '19877'
 '19928' '19943' '19947' '19950' '19952' '19972' '19988' '19996' '2003'
 '211536' '21440' '218629' '219533' '220367' '220845' '2223' '223596'
 '226593' '226875' '228414' '229236' '230080' '230136' '230

Hay mucha tipología distinta. Puedes observar que hay clasificación formada principalmente por un prefijo y seguida por una numeración.

Los prefijos son, entre otros: 


*   A/5 o A/4.
*   C + número
*   C.A.
*   PC
*   SOTON o STON


Filtraremos solo por el grupo de los que empiezan por PC (Private Cabin) para observar si la numeración sigue un orden.

In [ ]:
# Filtro para ver los que empiezan por PC 
filtro_billetes =[col for col in billetes if col.startswith('PC')!=0]

print(filtro_billetes)

['PC 17318', 'PC 17473', 'PC 17474', 'PC 17475', 'PC 17476', 'PC 17477', 'PC 17482', 'PC 17483', 'PC 17485', 'PC 17558', 'PC 17569', 'PC 17572', 'PC 17582', 'PC 17585', 'PC 17590', 'PC 17592', 'PC 17593', 'PC 17595', 'PC 17596', 'PC 17597', 'PC 17599', 'PC 17600', 'PC 17601', 'PC 17603', 'PC 17604', 'PC 17605', 'PC 17608', 'PC 17609', 'PC 17610', 'PC 17611', 'PC 17612', 'PC 17754', 'PC 17755', 'PC 17756', 'PC 17757', 'PC 17758', 'PC 17759', 'PC 17760', 'PC 17761']


Comprueba que hay saltos en la numeración, es decir no es una numeración consecutiva aunque a veces si se mantiene. Esto es, el primer elemento es PC 17318 y salta al PC 17473, luego sigue un orden hasta el PC 17477 donde se produce otro salto al 17482.

Claramente, sigue un "orden". No sigue un orden convencional seguramente debido a la clasificación del coste del billete asociado al tipo de cabina asociado.

<h4><strong>Análisis de la variable Cabin </strong></h4>

In [ ]:
# Ordenamos y quitamos valores nulos a Cabin
cabina = pd.unique(df['Cabin'].dropna())
cabina.sort()
print(cabina)

['A10' 'A14' 'A16' 'A19' 'A20' 'A23' 'A24' 'A26' 'A31' 'A32' 'A34' 'A36'
 'A5' 'A6' 'A7' 'B101' 'B102' 'B18' 'B19' 'B20' 'B22' 'B28' 'B3' 'B30'
 'B35' 'B37' 'B38' 'B39' 'B4' 'B41' 'B42' 'B49' 'B5' 'B50' 'B51 B53 B55'
 'B57 B59 B63 B66' 'B58 B60' 'B69' 'B71' 'B73' 'B77' 'B78' 'B79' 'B80'
 'B82 B84' 'B86' 'B94' 'B96 B98' 'C101' 'C103' 'C104' 'C106' 'C110' 'C111'
 'C118' 'C123' 'C124' 'C125' 'C126' 'C128' 'C148' 'C2' 'C22 C26'
 'C23 C25 C27' 'C30' 'C32' 'C45' 'C46' 'C47' 'C49' 'C50' 'C52' 'C54'
 'C62 C64' 'C65' 'C68' 'C7' 'C70' 'C78' 'C82' 'C83' 'C85' 'C86' 'C87'
 'C90' 'C91' 'C92' 'C93' 'C95' 'C99' 'D' 'D10 D12' 'D11' 'D15' 'D17' 'D19'
 'D20' 'D21' 'D26' 'D28' 'D30' 'D33' 'D35' 'D36' 'D37' 'D45' 'D46' 'D47'
 'D48' 'D49' 'D50' 'D56' 'D6' 'D7' 'D9' 'E10' 'E101' 'E12' 'E121' 'E17'
 'E24' 'E25' 'E31' 'E33' 'E34' 'E36' 'E38' 'E40' 'E44' 'E46' 'E49' 'E50'
 'E58' 'E63' 'E67' 'E68' 'E77' 'E8' 'F E69' 'F G63' 'F G73' 'F2' 'F33'
 'F38' 'F4' 'G6' 'T']


<p> <mark>SABÍAS QUE</mark> </p>
<hr>

La función de cabina es un poco complicada y necesita un poco más de exploración. Falta la gran parte de la función _Cabin_ y la función en sí no puede ignorarse por completo porque algunas de las cabinas pueden tener tasas de supervivencia más altas. 

La primera letra de los valores de _Cabin_ son las cubiertas en las que se encuentran las cabinas. Esas cubiertas se separaron, principalmente, para una clase de pasajeros, pero algunas de ellas fueron utilizadas por múltiples clases de pasajeros.

<br>

<img src="https://vignette.wikia.nocookie.net/titanic/images/f/f9/Titanic_side_plan.png/revision/latest?cb=20180322183733" alt="alt text"></p>
<ul>
<li>On the Boat Deck, there were <strong>6</strong> rooms labeled as <strong>T, U, W, X, Y, Z</strong>, but only the <strong>T</strong> cabin is present in the dataset.</li>
<li><strong>A</strong>, <strong>B</strong> and <strong>C</strong> decks were only for 1st class passengers.</li>
<li><strong>D</strong> and <strong>E</strong> decks were for all classes.</li>
<li><strong>F</strong> and <strong>G</strong> decks were for both 2nd and 3rd class passengers.</li>
<li>From going <strong>A</strong> to <strong>G</strong>, distance to the staircase increases which might be a factor of survival.</li>
</ul>

<center>
<figcaption>Imagen4. Estructura de la cabina del Titanic </figcaption>
</center>

Fuente de la imagen: [www.vignette.wikia.nocookie.net](https://vignette.wikia.nocookie.net/titanic/images/f/f9/Titanic_side_plan.png/revision/latest?cb=20180322183733)




<h4> <strong>Análisis de la variable Age </strong> </h4>

In [ ]:
edad = pd.unique(df['Age'])
edad.sort()
print(edad)

[ 0.42  0.67  0.75  0.83  0.92  1.    2.    3.    4.    5.    6.    7.
  8.    9.   10.   11.   12.   13.   14.   14.5  15.   16.   17.   18.
 19.   20.   20.5  21.   22.   23.   23.5  24.   24.5  25.   26.   27.
 28.   28.5  29.   30.   30.5  31.   32.   32.5  33.   34.   34.5  35.
 36.   36.5  37.   38.   39.   40.   40.5  41.   42.   43.   44.   45.
 45.5  46.   47.   48.   49.   50.   51.   52.   53.   54.   55.   55.5
 56.   57.   58.   59.   60.   61.   62.   63.   64.   65.   66.   70.
 70.5  71.   74.   80.     nan]


**La edad, en general, se suele representar como variable cuantitativa discreta**, esto es, con números enteros (1,2,3,...) sin valorar los meses (días, horas, minutos o segundos).

Aunque en algunos estudios resulta relevante, como en las compañías aseguradoras cuando trabajan seguros de vida que miden milimétricamente el riesgo a asumir, y, por tanto, en ellas no se considera igual a una persona de 70 años que a otra con 70 años y 6 meses.

Como decíamos, las compañías aseguradoras miden el riesgo, luego una persona con 70 años y 6 meses tiene más probabilidades de ocurrencia que una persona recién cumplidos los 70 años que presenta menor riesgo. Cada vez el mercado es más competitivo, por lo que las compañías "ajustan" y miden mejor.

Lo normal es que en este tipo de mediciones, se busque mayor precisión y, por ende, **la edad será una variable cuantitativa de tipo continuo** (contemplando unidades decimales que podrán representar los meses o días).

Por tanto, no sólo hay que analizar la naturaleza de la información, si no también cómo está recogida dicha información. En general, será más fácil que las variables de tipo discreto estén alineadas.

Por ejemplo, nadie pondría un hijo y medio. Normalmente, en estadística, se estima al alza, por tanto, si sale en promedio un hijo y medio se pondrá como resultado 2 hijos. 

Aunque la naturaleza de la información influye en el tipo de variable, no es determinante. Así pues, la manera en la que se recoge el dato influye en el tipo de variable. Un caso en el que se ve claramente son las herramientas de medición (sonidos, temperaturas, balanzas, etc.) puesto que pueden ser más o menos precisas. 

Por ejemplo, la temperatura es una variable de tipo continuo, pero puede ocurrir que el aparato de medición no tenga suficiente precisión, por lo que al almacenar la información sólo contenga unidades enteras (30º, 35º, etc.) en lugar de unidades decimales (30.5º, 35.33º, etc). En este caso, la variable **temperatura en origen es de tipo cuantitativa continua** aunque la toma del dato ha **transformado la variable a una cuantitativa de tipo discreto.**

\\
<p> <mark>IMPORTANTE</mark> </p>
<hr>

Como norma general, _**se tratan los datos como se han recogido**_, aunque siempre conviene hacer una mención y explicar cuando ocurran ese tipo de "incongruencias".

Desde el punto de vista del análisis de datos, el **tipo de variables** sería:

\\
\begin{equation} 
Variables = 
 \begin{cases}
   \text{Clasificación: } \text{Pclass, Name, Sex, Age, Ticket, Cabin, Embarked, Survived.} \\
    \\
   \text{Analíticas: } \text{SibSp, Parch, Fare, Survived (sólo conteo). }  \\ 
  \end{cases}\\     
\end{equation}

\\
<p> <mark>OBSERVACIONES</mark> </p>
<hr>


*  _**PassengerId:**_ dato que es usado para agilizar el cruce de información y que, al mismo tiempo, no aporta información para el análisis de los datos.

*  _**Survived:**_ al ser una variable dicotómica (sobrevive si o no), su uso principal es de tipo clasificatorio, pero también se puede usar de manera analítica (como por ejemplo para realizar un conteo simple que sirva para saber la proporción de supervivientes que hubo dentro de toda la población).

# **IDEAS CLAVE**
<br>
<hr>
<p> <h1> <center> <strong> Estadística </center> </strong> </h1> </p>
<hr>
<br> 

*   **La estadística** es el estudio que **recuenta, clasifica y analiza todos los hechos** que tienen una determinada característica en común, para poder **llegar a conclusiones a partir de los datos** numéricos extraídos.

*   La estadística **es una herramienta que utilizamos para la toma de decisiones** en cada uno de los campos de nuestro entorno social, por tanto es una materia importante dentro de nuestro proceso de aprendizaje.

*   Es imprescindible analizar la información **antes de profundizar en cualquier tipo de análisis** y, para ello,  **el primer paso consiste en analizar el tipo de variables** con el que trabajaremos.

